test/rule-matches/label-content-name-mismatch-matches.js

```diff
@@ -0,0 +1,170 @@
+			'<button name="link">Until the very end.</button>',
+			'button'
+		);
+		var actual = rule.matches(target);
+		assert.isFalse(actual);
+	});
+
+	it('returns true if element has accessible name (`aria-label`)', function() {
+		var target = fixtureSetup(
+			'<button name="link" aria-label="Until">Until the very end.</button>',
+			'button'
+		);
+		var actual = rule.matches(target);
+		assert.isTrue(actual);
+	});
+
+	it('returns false if element has empty accessible name (`aria-label`)', function() {
+		var target = fixtureSetup(
+			'<button name="link" aria-label="">Until the very end.</button>',
+			'button'
+		);
+		var actual = rule.matches(target);
+		assert.isFalse(actual);
+	});
+
+	it('returns true if element has accessible name (`aria-labelledby`)', function() {
+		var target = fixtureSetup(
+			'<div role="button" id="target" aria-labelledby="foo">some content</div>' +
+				'<div id="foo">Foo text</div>',
+			'div#target'
+		);
+		var actual = rule.matches(target);
+		assert.isTrue(actual);
+	});
+
+	it('returns false if element has empty accessible name (`aria-labelledby`)', function() {
+		var target = fixtureSetup(
+			'<div role="button" id="target" aria-labelledby="foo">some content</div>' +
+				'<div id="foo"></div>',
+			'div#target'
+		);
+		var actual = rule.matches(target);
+		assert.isFalse(actual);
+	});
+
+	it('returns false if element has empty accessible name (`aria-labelledby`) because idref does not exist', function() {
+		var target = fixtureSetup(
+			'<div role="button" id="target" aria-labelledby="doesNotExist">some content</div>' +
+				'<div id="idExists">Right Label</div>',
+			'div#target'
+		);
+		var actual = rule.matches(target);
+		assert.isFalse(actual);
+	});
+
+	it('returns true if element has accessible name (`aria-labelledby`) - multiple refs', function() {
+		var target = fixtureSetup(
+			'<div role="button" id="target" aria-labelledby="bar baz foo">some content</div>' +
+				'<div id="foo">Foo</div>' +
+				'<div id="bar">Bar</div>' +
+				'<div id="baz">Baz</div>',
+			'div#target'
+		);
+		var actual = rule.matches(target);
+		assert.isTrue(actual);
+	});
+
+	it('returns false for `non-widget` role', function() {
+		var target = fixtureSetup('<a aria-label="OK">Next</a>', 'a');
+		var actual = rule.matches(target);
+		assert.isFalse(actual);
+	});
+
+	it('returns false for widget role that does not support name from content', function() {
+		var target = fixtureSetup(
+			'<input type="email" aria-label="E-mail" value="Contact">',
+			'input'
+		);
+		var actual = rule.matches(target);
+		assert.isFalse(actual);
+	});
+
+	it('returns false for non-widget role that does support name from content', function() {
+		var target = fixtureSetup(
+			'<div role="doc-chapter" aria-label="OK">Next</div>',
+			'div'
+		);
+		var actual = rule.matches(target);
+		assert.isFalse(actual);
+	});
+
+	it('returns true for non-widget role that does support name from content', function() {
```

Comment:
Doesn't the rule say that it should only test widgets? Why are we testing this node?